src/tax-lot/tax-lot.service.ts

```diff
@@ -1,35 +1,152 @@
 import { InjectRepository } from "@mikro-orm/nestjs";
 import { Inject, Injectable } from "@nestjs/common";
 import { TaxLot } from "./tax-lot.entity";
+import { DB, DbType } from "src/global/providers/db.provider";
 import { TaxLotRepository } from "./tax-lot.repository";
 import { FeatureFlagConfig } from "src/config";
 import { ConfigType } from "@nestjs/config";
+import {
+  taxLot,
+  zoningDistrict,
+  zoningDistrictClass,
+  zoningDistrictZoningDistrictClass,
+} from "src/schema";
+import { eq, sql } from "drizzle-orm";
 
 @Injectable()
 export class TaxLotService {
   constructor(
     @InjectRepository(TaxLot)
     private readonly taxLotRepository: TaxLotRepository,
 
+    @Inject(DB)
+    private readonly db: DbType,
+
     @Inject(FeatureFlagConfig.KEY)
     private featureFlagConfig: ConfigType<typeof FeatureFlagConfig>,
   ) {}
 
-  async findTaxLotByBbl(bbl: string): Promise<TaxLot | null> {
-    if (this.featureFlagConfig.useDrizzle)
-      throw new Error("Tax lot route not support in drizzle");
-    return this.taxLotRepository.findOne(
-      { bbl },
-      {
-        fields: ["bbl", "borough", "landUse", "block", "lot", "address"],
-        populate: ["borough", "landUse"],
-      },
-    );
+  async findTaxLotByBbl(bbl: string) {
+    if (this.featureFlagConfig.useDrizzle) {
+      const result = await this.db.query.taxLot.findFirst({
+        columns: {
+          boroughId: false,
+          landUseId: false,
+          wgs84: false,
+          liFt: false,
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+      return result !== undefined ? result : null;
+    } else {
+      return this.taxLotRepository.findOne(
+        { bbl },
+        {
+          fields: ["bbl", "borough", "landUse", "block", "lot", "address"],
+          populate: ["borough", "landUse"],
+        },
+      );
+    }
+  }
+
+  async findTaxLotByBblGeoJson(bbl: string) {
+    if (this.featureFlagConfig.useDrizzle) {
+      const result = await this.db.query.taxLot.findFirst({
+        columns: {
+          bbl: true,
+          block: true,
+          lot: true,
+          address: true,
+        },
+        extras: {
+          geometry: sql<string>`ST_AsGeoJSON(${taxLot.wgs84}, 6)`.as(
+            "geometry",
+          ),
+        },
+        where: (taxLot, { eq }) => eq(taxLot.bbl, bbl),
+        with: {
+          borough: true,
+          landUse: true,
+        },
+      });
+      if (result === undefined) return null;
+      const geometry = JSON.parse(result.geometry);
```

Comment:
@TangoYankee  Did you try implementing Drizzle's `toDriver` and `fromDriver` [methods](https://orm.drizzle.team/docs/custom-types#ts-doc-for-type-definitions) in the postgis column custom types to do this parsing implicitly? It seems to me like this would be a good use case for that. If I'm understanding it correctly, it would eliminate the need to call `ST_AsGeoJSON` in `extras` and the need for this parsing in the Service.

Reply:
I had a hiccup when I tried in purple-hawk. But, I'll give it another attempt.